ocaml/xapi/extauth_plugin_ADwinbind.ml

```diff
@@ -227,9 +253,38 @@ module Ldap = struct
             args
         in
         Ok stdout
-      with _ -> Error (generic_ex "ldap query failed")
+      with _ -> Error (generic_ex "ldap query user info from sid failed")
     in
     parse_user stdout <!> generic_ex "%s"
+
+  let query_sid ~name ~kdc ~domain_netbios =
+    let key = "objectSid" in
+    let env = env_of_lookup domain_netbios in
+    let query = Printf.sprintf "(|(sAMAccountName=%s)(name=%s))" name name in
```

Comment:
doesn't this need escaping? It looks like someone can inject arbitrary values into the LDAP query this way if they add the right number of `)` into the login name.

Reply:
I will escape the name, but I do not lose my sleep with it as
# This only happens when the winbind cache is missed, winbind cache is refreshed every 60 seconds.
# This just query the subject SID, and does not update system,  Add subject and change subject role will need pool-admin permission.